core/src/main/java/org/lflang/federated/generator/FedGenerator.java

```diff
@@ -416,6 +416,13 @@ private void analyzeFederates(Reactor federation, LFGeneratorContext context) {
         && !federation.getHost().getAddr().equals("localhost")) {
       rtiConfig.setHost(federation.getHost().getAddr());
     }
+    // If the federation is dockerized, use "rti" as the hostname.
+    // FIXME: Do we have to check the condition rtiConfig.getHost().equals("localhost")?
```

Comment:
This is a good question. For a federation that runs on a single machine, using `localhost` works just fine, but this doesn't work for Docker because it creates processes in their own container, each of which has their own distinct "localhost." I think your fix is correct.

Reply:
Thanks for answering, @lhstrh!